ginga/misc/plugins/MultiDim.py

```diff
@@ -254,6 +260,27 @@ def build_naxis(self, dims):
         # Add vbox of naxis controls to gui
         self.naxisfr.set_widget(w)
 
+        # for storing play_idx for each dim of image. used for going back to
+        # the idx where you left off.
+        self.play_indices = [0 for i in range(len(dims) - 2)]
+
+        def play_axis_change_func_creator(n):
+            def play_axis_change():
+                # print "play_axis changed to %d" % n
+                # print "play_idx switch from %d to %d" %(self.play_indices[self.play_axis-2], self.play_indices[n-2])
+                self.play_indices[self.play_axis - 2] = self.play_idx
+                self.play_axis = n
+                if self.play_axis < len(dims):
+                    self.play_max = dims[self.play_axis]
+                self.play_idx = self.play_indices[n - 2]
+            return play_axis_change
+
+        for n in range(2, len(dims)):
+            key = 'axis%d' % (n + 1)
+            # the following line doesn't trigger, hence, the hack in the following line
+            # self.w[key].set_callback('activated', play_axis_change_func_creator(n) )
+            self.w[key].widget.toggled.connect(play_axis_change_func_creator(n))
```

Comment:
For some reason, `set_callback` doesn't activate the function given. Instead, for proof of concept, I used a hack that only works for the qtw radiobutton.